semsim/process_ontology.py

```diff
@@ -0,0 +1,62 @@
+"""Process ontology and retrieve pairwise similarities."""
+from collections import Counter
+
+import pandas as pd
+
+from .compute_pairwise_similarities import (compute_pairwise_ancestors_jaccard,
+                                            compute_pairwise_resnik)
+
+GRAPE_DATA_MOD = "grape.datasets.kgobo"
+
+
+def get_similarities(
+    ontology: str, resnik_path: str, ancestors_jaccard_path: str
+):
+    """Compute and store similarities to the provided paths.
+
+    :param ontology: str, name of ontology to retrieve and process.
+    :param resnik_path: str, where to store the resnik pairwise similarities.
+    :param ancestors_jaccard_path: str, where to store the Ancestors Jaccard
+        pairwise similarities.
+    """
+    ontology = ontology.upper()
+    onto_graph_class = import_grape_class(ontology)
+
+    onto_graph = (
+        onto_graph_class(directed=True)
+        .filter_from_names(
+            edge_type_names_to_keep=["biolink:subclass_of"],
+            node_prefixes_to_keep=[f"{ontology}:"],
+        )
+        .to_transposed()
+        .remove_disconnected_nodes()
+    )
+
+    counts = dict(
+        Counter(
+            pd.read_csv(
+                "http://purl.obolibrary.org/obo/hp/hpoa/phenotype.hpoa",
```

Comment:
Should we have URIs declared as a variable (or a separate `constants.py` file?)

Reply:
[ontologies.yaml](https://obofoundry.org/registry/ontologies.yml) as Harry suggested earlier.

Reply:
✅

Reply:
The ontologies can be derived from OBO Foundry's ontologies.yaml, but unfortunately they don't include links to the annotations like phenotype.hpoa.